pkg/reconciler/kubernetes/tektonchain/tektonchain.go

```diff
@@ -678,46 +676,19 @@ func AddControllerEnv(controllerEnvs []corev1.EnvVar) mf.Transformer {
 	}
 }
 
-func GenerateSigningSecrets(ctx context.Context) map[string][]byte {
+func generateSigningSecrets(ctx context.Context) map[string][]byte {
 	logger := logging.FromContext(ctx)
-	// Generate ECDSA key pair
-	privateKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
-	if err != nil {
-		logger.Error("Error generating private key:", err)
-		return nil
-	}
 
-	// Convert private key to PKCS8
-	privateKeyBytes, err := x509.MarshalPKCS8PrivateKey(privateKey)
+	// Here, nil indicates that no password is provided for the private key encryption, as it requires a user-provided secret.
+	keys, err := cosign.GenerateKeyPair(nil)
```

Comment:
Could we generate a random password on the user's behalf?

Reply:
Okay, makes sense. Added.